rpcs3/rpcs3qt/instruction_editor_dialog.cpp

```diff
@@ -13,17 +13,33 @@ constexpr auto qstr = QString::fromStdString;
 
 extern bool ppu_patch(u32 addr, u32 value);
 
-instruction_editor_dialog::instruction_editor_dialog(QWidget *parent, u32 _pc, const std::shared_ptr<cpu_thread>& _cpu, CPUDisAsm* _disasm)
+instruction_editor_dialog::instruction_editor_dialog(QWidget *parent, u32 _pc, CPUDisAsm* _disasm, std::function<cpu_thread*()> func, instruction_editor_dialog** storage_pointer)
 	: QDialog(parent)
 	, m_pc(_pc)
 	, m_disasm(_disasm)
-	, m_cpu(_cpu)
+	, m_get_cpu(std::move(func))
 {
 	setWindowTitle(tr("Edit instruction"));
 	setAttribute(Qt::WA_DeleteOnClose);
 	setMinimumSize(300, sizeHint().height());
 
-	m_cpu_offset = m_cpu->id_type() == 2 ? static_cast<spu_thread&>(*m_cpu).ls : vm::g_sudo_addr;
+	if (storage_pointer)
+	{
+		connect(this, &QDialog::finished, [storage_pointer](int)
+		{
+			*storage_pointer = nullptr;
```

Comment:
what are you trying to achieve here ?

Reply:
Setting m_kernel_explorer, m_inst_editor and m_reg_editor to nullptr after the dialog is closed.

Reply:
why

Reply:
Because I check if they are nullptr.

Reply:
then use this and don't overcomplicate things...
connect(m_reg_editor, &QDialog::finished, this, [this]() { m_reg_editor = nullptr; });

Reply:
Done.